torch/utils/data/dataloader.py

```diff
@@ -272,38 +254,44 @@ class DataLoader(object):
             at every epoch (default: False).
         sampler (Sampler, optional): defines the strategy to draw samples from
             the dataset. If specified, the ``shuffle`` argument is ignored.
+        batch_sampler (Sampler, optional): like sampler, but returns a batch of
+            indices at a time. If specified, batch_size, shuffle, sampler, and
+            drop_last are ignored.
         num_workers (int, optional): how many subprocesses to use for data
             loading. 0 means that the data will be loaded in the main process
             (default: 0)
-        collate_fn (callable, optional)
-        pin_memory (bool, optional)
+        collate_fn (callable, optional): merges a list of samples to form a mini-batch.
+        pin_memory (bool, optional): If ``True``, the data loader will copy tensors
+            into CUDA pinned memory before returning them.
         drop_last (bool, optional): set to ``True`` to drop the last incomplete batch,
             if the dataset size is not divisible by the batch size. If False and
             the size of dataset is not divisible by the batch size, then the last batch
             will be smaller. (default: False)
     """
 
-    def __init__(self, dataset, batch_size=1, shuffle=False, sampler=None, num_workers=0,
-                 collate_fn=default_collate, pin_memory=False, drop_last=False):
+    def __init__(self, dataset, batch_size=1, shuffle=False, sampler=None, batch_sampler=None,
```

Comment:
I think it's unlikely to break anyone in this case and it's nicer to have it next to sampler in the docs.